File: a_first_lesson/src/subscriber.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String


def callback(message: String):
    rospy.loginfo(rospy.get_caller_id() + "I heard %s", message.data)


def subscriber():
    rospy.init_node("subscriber_node")
    rospy.Subscriber("subscribe", String, callback=callback)
    rospy.spin()


if __name__ == '__main__':
    subscriber()

File: a_first_lesson/src/subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void callback(const std_msgs::String::ConstPtr &msg) {
    ROS_INFO("[subscriber] I heard [%s]\n", msg->data.c_str());
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "subscriber_node");
    ros::NodeHandle node;

    ros::Subscriber subscriber = node.subscribe("publisher", 1000, callback);

    ros::spin();

    return 0;
}